drivers/hmis/spec/requests/hmis/lookup_client_spec.rb

```diff
@@ -175,6 +176,55 @@
     expect(result.dig('data', 'client', 'image')).not_to be_nil
   end
 
+  context 'with a client who is chronically homeless per HUD definition' do
+    let!(:warehouse_ds) { create :destination_data_source }
+    let!(:client) { create :grda_warehouse_hud_client, data_source_id: warehouse_ds.id }
```

Comment:
@martha we have a factory `hmis_hud_client_with_warehouse_client` that _should_ help with this setup. I believe it will generate the source client, warehouse client, and the destination client.

I think it just needs some tweaks, because it looks like it generates the destination client in the same ds as the source, which isn't right.  A potential modification for https://github.com/greenriver/hmis-warehouse/blob/7f1164546483039af1c9c479f1df81e10de817d5/drivers/hmis/spec/factories/hmis/hud/clients.rb#L60-L71

is:
```ruby
  # Client that is in the Warehouse destination data source (not the HMIS data source)
  factory :destination_client, parent: :hmis_hud_base_client do
    data_source { association :destination_data_source }
  end

  # WarehouseClient that links the source HMIS client to the destination client
  factory :hmis_warehouse_client, class: 'Hmis::WarehouseClient' do
    data_source { association :hmis_data_source }
    destination { association :destination_client }
    source { association :hmis_hud_base_client, data_source: data_source }
    sequence(:id_in_source, 100)
  end

  # HMIS Source Client that is linked to a destination client (Via WarehouseClient)
  factory :hmis_hud_client_with_warehouse_client, parent: :hmis_hud_base_client do
    after(:create) do |client|
      create(:hmis_warehouse_client, data_source: client.data_source, source: client)
    end
  end
```

If that works, I thiiiink you should be able to keep the setup simple here:
```
let!(:client) {  create :hmis_hud_client_with_warehouse_client, data_source: ds1 }
let!(:enrollment) {  create :hmis_hud_enrollment, data_source: ds1, project: p1, client: client }
```

Reply:
Thanks! This is interesting, but I think that the factories need to be further modified so that the two clients have the same Personal Id? I will keep poking to see if I can find out how to do that in factory bot.